xmtp/src/conversations.rs

```diff
@@ -189,12 +195,44 @@ where
             }
         }
 
-        self.send_outbound_payloads().await;
         Ok(())
     }
 
-    // TODO push payloads onto the network
-    pub(crate) async fn send_outbound_payloads(&self) {}
+    pub async fn publish_outbound_payloads(&self) -> Result<(), ConversationError> {
+        let unsent_payloads = self.client.store.fetch_and_lock_outbound_payloads(
+            OutboundPayloadState::Pending,
+            Duration::from_secs(60).as_nanos() as i64,
+        )?;
+
+        if unsent_payloads.is_empty() {
+            return Ok(());
+        }
+
+        let payload_ids = unsent_payloads
+            .iter()
+            .map(|payload| payload.created_at_ns)
+            .collect();
+        let envelopes = unsent_payloads
+            .into_iter()
```

Comment:
SAND: Moving the Envelopes in this context doesn't seem necessary. A reference would be sufficient and easier to understand. To keep a more consistent memory model using `iter` should be favored over `into_iter` unless there is a clear requirement